source/extensions/common/dynamic_forward_proxy/dns_cache_impl.h

```diff
@@ -186,10 +192,12 @@ class DnsCacheImpl : public DnsCache, Logger::Loggable<Logger::Id::forward_proxy
   PrimaryHostInfo& getPrimaryHost(const std::string& host);
 
   void addCacheEntry(const std::string& host,
-                     const Network::Address::InstanceConstSharedPtr& address);
+                     const Network::Address::InstanceConstSharedPtr& address,
+                     const std::chrono::seconds& ttl);
```

Comment:
I think normally we'd pass this by value since it's just a wrapper around an integer